test/TestUtilities/Test.Utility/TestDotnetCLiUtility.cs

```diff
@@ -192,36 +195,39 @@ private static void UpdateCliWithLatestNuGetAssemblies(string cliDirectory)
-            const string restoreTargetsName = "NuGet.targets";
-            const string restoreTargetsExtName = "NuGet.RestoreEx.targets";
+            var fileExtensions = new[] { ".dll", ".pdb", ".targets" };
 
-            var sdkDependencies = new List<string> { restoreProjectName, "NuGet.Versioning", "NuGet.Protocol", "NuGet.ProjectModel", "NuGet.Packaging", "NuGet.LibraryModel", "NuGet.Frameworks", "NuGet.DependencyResolver.Core", "NuGet.Configuration", "NuGet.Common", "NuGet.Commands", "NuGet.CommandLine.XPlat", "NuGet.Credentials", "NuGet.Build.Tasks.Console" };
+            var sdkDependencies = new List<string> { "NuGet.Build.Tasks.Console", "NuGet.CommandLine.XPlat" };
 
-            // Copy rest of the NuGet assemblies.
             foreach (var projectName in sdkDependencies)
             {
                 var projectArtifactsBinFolder = Path.Combine(artifactsDirectory, projectName, "bin", configuration);
 
                 var tfmToCopy = GetTfmToCopy(projectArtifactsBinFolder);
                 var frameworkArtifactsFolder = new DirectoryInfo(Path.Combine(projectArtifactsBinFolder, tfmToCopy));
 
-                var fileName = projectName + ".dll";
-                File.Copy(
-                        sourceFileName: Path.Combine(frameworkArtifactsFolder.FullName, fileName),
-                        destFileName: Path.Combine(pathToSdkInCli, fileName),
-                        overwrite: true);
-                // Copy the restore targets.
-                if (projectName.Equals(restoreProjectName))
+                foreach (string fileExtension in fileExtensions)
                 {
-                    File.Copy(
-                        sourceFileName: Path.Combine(frameworkArtifactsFolder.FullName, restoreTargetsName),
-                        destFileName: Path.Combine(pathToSdkInCli, restoreTargetsName),
-                        overwrite: true);
-                    File.Copy(
-                        sourceFileName: Path.Combine(frameworkArtifactsFolder.FullName, restoreTargetsExtName),
-                        destFileName: Path.Combine(pathToSdkInCli, restoreTargetsExtName),
-                        overwrite: true);
+                    foreach (FileInfo file in frameworkArtifactsFolder.EnumerateFiles($"*{fileExtension}"))
+                    {
+                        file.CopyTo(Path.Combine(pathToSdkInCli, file.Name), overwrite: true);
+                    }
+                }
+
+                // Update NuGet.CommandLine.XPlat.runtimeconfig.json so that our dependencies are loaded correctly
+                FileInfo runtimeConfigJsonFile = new FileInfo(Path.Combine(pathToSdkInCli, $"{projectName}.runtimeconfig.json"));
+
+                JObject jObject = runtimeConfigJsonFile.Exists ? JsonUtility.LoadJson(runtimeConfigJsonFile.OpenText()) : new JObject();
+
+                jObject["runtimeOptions"]["configProperties"]["Microsoft.NETCore.DotNetHostPolicy.SetAppPaths"] = true;
```

Comment:
May I know what's this flag for? Is there any document about it? Thanks!

Reply:
It took me a while to find it, I don't think the .NET folks want people knowing about it or using it.  We are only allowed to use it for testing so we're okay though.  I added our team to an email thread about how I found this and what it does.